### PR TITLE
[java] ClassNamingConventions - add new property testClassPattern

### LINKED ISSUE #2867: [java] Separate pattern for test classes in ClassNamingConventions rule for Java
Apex has `testClassPattern` - https://pmd.github.io/latest/pmd_rules_apex_codestyle.html#classnamingconventions

Java doesn't - https://pmd.github.io/latest/pmd_rules_java_codestyle.html#classnamingconventions

Having test classes names consistently end with "Test" is important to make sure they are correctly picked up by Maven Surefire plugin. I would say this is even more important than following other naming conventions.

### PR DESCRIPTION
**Pull request by adangel**
## Related issues

- Fixes #2867
- Refs https://github.com/pmd/pmd/discussions/4179

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [x] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 2177 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/c44779a4a96045ce86e6869535138104/diff1/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/ClassNamingConventionsRule.java [hunk 1]
```diff
@@ -31,6 +33,9 @@ public class ClassNamingConventionsRule extends AbstractNamingConventionRule<AST
     private final PropertyDescriptor<Pattern> enumerationRegex = defaultProp("enum").build();
     private final PropertyDescriptor<Pattern> annotationRegex = defaultProp("annotation").build();
     private final PropertyDescriptor<Pattern> utilityClassRegex = defaultProp("utility class").build();
+    private final PropertyDescriptor<Pattern> testClassRegex = defaultProp("test class")
+            .desc("Regex which applies to test class names. Since PMD 6.52.0.")
+            .defaultValue("[A-Z][a-zA-Z0-9]*Test").build();
```
Comment (adangel) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/ClassNamingConventionsRule.java:38]: We probably should change the pattern:

> The class name 'AfterAdviceFirstTests' doesn't match '[A-Z][a-zA-Z0-9]*Test'

```suggestion
            .defaultValue("[A-Z][a-zA-Z0-9]*Tests?").build();
```

File: pmd-java/src/main/resources/category/java/codestyle.xml [hunk 2]
```diff
@@ -408,6 +408,10 @@ public class Foo extends Bar{
             For this rule, a utility class is defined as: a concrete class that does not
             inherit from a super class or implement any interface and only has static fields
             or methods.
+
+            This rule detects test classes using the following convention: Test classes are classes, that
```
Comment (oowekyala) [inline pmd-java/src/main/resources/category/java/codestyle.xml:412]:
```suggestion
            This rule detects test classes using the following convention: Test classes are top-level classes, that
```

File: docs/pages/release_notes.md [hunk 3]
```diff
@@ -14,11 +14,19 @@ This is a {{ site.pmd.release_type }} release.
 
 ### New and noteworthy
 
+#### Modified rules
+
+* The rule {% rule java/codestyle/ClassNamingConventions %} has a new property `testClassPattern`, which is applied
+  to test classes. By default, test classes should end with the suffix "Test". Test classes are classes, that
```
Comment (oowekyala) [inline docs/pages/release_notes.md:20]:
```suggestion
  to test classes. By default, test classes should end with the suffix "Test". Test classes are top-level classes, that
```

**Review by oowekyala — APPROVED** [review]
Thanks!

**Pull request merged**